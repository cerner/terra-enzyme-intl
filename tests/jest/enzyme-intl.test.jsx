import React from 'react';
import { shallow, mount } from 'enzyme';
import { FormattedMessage, injectIntl } from 'react-intl';
import {
  shallowWithIntl,
  mountWithIntl,
  renderWithIntl,
  mockIntl,
  shallowContext,
  mountContext,
} from '../../src';

const prettyPrintProps = props => JSON.stringify(props, undefined, 2);

const FormattedMessageSubject = props => (
  <div>
    <FormattedMessage id="TerraEnzymeIntl.helloWorld" />
    <FormattedMessage id="TerraEnzymeIntl.buttonText">
      {text => <button type="button">{text}</button>}
    </FormattedMessage>
    {prettyPrintProps(props)}
  </div>
);

const InjectIntlSubject = injectIntl(({ intl, ...props }) => (
  <div>
    {intl.formatMessage({ id: 'TerraEnzymeIntl.helloWorld' })}
    <button type="button">{intl.formatMessage({ id: 'TerraEnzymeIntl.buttonText' })}</button>
    {prettyPrintProps(props)}
  </div>
));

const ContextTypesSubject = (props, { intl }) => (
  <div>
    {intl.formatMessage({ id: 'TerraEnzymeIntl.helloWorld' })}
    <button type="button">{intl.formatMessage({ id: 'TerraEnzymeIntl.buttonText' })}</button>
    {prettyPrintProps(props)}
  </div>
);
ContextTypesSubject.contextTypes = {
  /* eslint-disable consistent-return */
  intl: (context) => {
    if (context.intl === undefined) {
      return new Error('Component is internationalized, and must be wrapped in terra-base');
    }
  },
  /* eslint-enable consistent-return */
};

describe('shallowWithIntl', () => {
  describe('using FormattedMessage', () => {
    const subject = <FormattedMessageSubject foo="bar" />;
    const wrapper = shallowWithIntl(subject);

    it('should match the snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('using injectIntl', () => {
    const subject = <InjectIntlSubject foo="bar" />;
    const wrapper = shallowWithIntl(subject);

    it('should match the snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('using contextTypes', () => {
    const subject = <ContextTypesSubject foo="bar" />;
    const wrapper = shallowWithIntl(subject);

    it('should match the snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});

describe('mountWithIntl', () => {
  describe('using FormattedMessage', () => {
    const subject = <FormattedMessageSubject foo="bar" />;
    const wrapper = mountWithIntl(subject);

    it('should match the snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('using injectIntl', () => {
    const subject = <InjectIntlSubject foo="bar" />;
    const wrapper = mountWithIntl(subject);

    it('should match the snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('using contextTypes', () => {
    const subject = <ContextTypesSubject foo="bar" />;
    const wrapper = mountWithIntl(subject);

    it('should match the snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});

describe('renderWithIntl', () => {
  describe('using FormattedMessage', () => {
    const subject = <FormattedMessageSubject foo="bar" />;
    const wrapper = renderWithIntl(subject);

    it('should match the snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('using injectIntl', () => {
    const subject = <InjectIntlSubject foo="bar" />;
    const wrapper = renderWithIntl(subject);

    it('should match the snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('using contextTypes', () => {
    const subject = <ContextTypesSubject foo="bar" />;
    const wrapper = renderWithIntl(subject);

    it('should match the snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});

describe('mockIntl', () => {
  const subject = intl => intl.formatMessage({ id: 'TerraEnzymeIntl.foo' });
  const expected = 'TerraEnzymeIntl.foo';

  describe('when used in a method expecting the react-intl intl object', () => {
    it('should match the expected output', () => {
      expect(subject(mockIntl)).toEqual(expected);
    });
  });
});


describe('shallowContext with shallow', () => {
  describe('using FormattedMessage', () => {
    const subject = <FormattedMessageSubject foo="bar" />;
    const wrapper = shallow(subject, shallowContext);

    it('should match the snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('using injectIntl', () => {
    const subject = <InjectIntlSubject foo="bar" />;
    const wrapper = shallow(subject, shallowContext);

    it('should match the snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('using contextTypes', () => {
    const subject = <ContextTypesSubject foo="bar" />;
    const wrapper = shallow(subject, shallowContext);

    it('should match the snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});

describe('mountContext with mount', () => {
  describe('using FormattedMessage', () => {
    const subject = <FormattedMessageSubject foo="bar" />;
    const wrapper = mount(subject, mountContext);

    it('should match the snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('using injectIntl', () => {
    const subject = <InjectIntlSubject foo="bar" />;
    const wrapper = mount(subject, mountContext);

    it('should match the snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('using contextTypes', () => {
    const subject = <ContextTypesSubject foo="bar" />;
    const wrapper = mount(subject, mountContext);

    it('should match the snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
