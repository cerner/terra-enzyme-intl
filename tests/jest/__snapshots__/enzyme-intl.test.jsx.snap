// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mountContext with mount using FormattedMessage should match the snapshot 1`] = `
<FormattedMessageSubject
  foo="bar"
>
  <div>
    <FormattedMessage
      id="TerraEnzymeIntl.helloWorld"
      values={Object {}}
    >
      <span>
        TerraEnzymeIntl.helloWorld
      </span>
    </FormattedMessage>
    <FormattedMessage
      id="TerraEnzymeIntl.buttonText"
      values={Object {}}
    >
      <button
        type="button"
      >
        TerraEnzymeIntl.buttonText
      </button>
    </FormattedMessage>
    {
  "foo": "bar"
}
  </div>
</FormattedMessageSubject>
`;

exports[`mountContext with mount using contextTypes should match the snapshot 1`] = `
<ContextTypesSubject
  foo="bar"
>
  <div>
    TerraEnzymeIntl.helloWorld
    <button
      type="button"
    >
      TerraEnzymeIntl.buttonText
    </button>
    {
  "foo": "bar"
}
  </div>
</ContextTypesSubject>
`;

exports[`mountContext with mount using injectIntl should match the snapshot 1`] = `
<InjectIntl(Component)
  foo="bar"
>
  <Component
    foo="bar"
    intl={
      Object {
        "defaultFormats": Object {},
        "defaultLocale": "en",
        "formatDate": [Function],
        "formatHTMLMessage": [Function],
        "formatMessage": [Function],
        "formatNumber": [Function],
        "formatPlural": [Function],
        "formatRelative": [Function],
        "formatTime": [Function],
        "formats": Object {},
        "formatters": Object {
          "getDateTimeFormat": [Function],
          "getMessageFormat": [Function],
          "getNumberFormat": [Function],
          "getPluralFormat": [Function],
          "getRelativeFormat": [Function],
        },
        "locale": "en",
        "messages": null,
        "now": [Function],
        "onError": [Function],
        "textComponent": "span",
        "timeZone": null,
      }
    }
  >
    <div>
      TerraEnzymeIntl.helloWorld
      <button
        type="button"
      >
        TerraEnzymeIntl.buttonText
      </button>
      {
  "foo": "bar"
}
    </div>
  </Component>
</InjectIntl(Component)>
`;

exports[`mountWithIntl using FormattedMessage should match the snapshot 1`] = `
<FormattedMessageSubject
  foo="bar"
  intl={
    Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "formats": Object {},
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralFormat": [Function],
        "getRelativeFormat": [Function],
      },
      "locale": "en",
      "messages": null,
      "now": [Function],
      "onError": [Function],
      "textComponent": "span",
      "timeZone": null,
    }
  }
>
  <div>
    <FormattedMessage
      id="TerraEnzymeIntl.helloWorld"
      values={Object {}}
    >
      <span>
        TerraEnzymeIntl.helloWorld
      </span>
    </FormattedMessage>
    <FormattedMessage
      id="TerraEnzymeIntl.buttonText"
      values={Object {}}
    >
      <button
        type="button"
      >
        TerraEnzymeIntl.buttonText
      </button>
    </FormattedMessage>
    {
  "foo": "bar",
  "intl": {
    "locale": "en",
    "messages": {},
    "formats": {},
    "timeZone": null,
    "textComponent": "span",
    "defaultLocale": "en",
    "defaultFormats": {},
    "formatters": {}
  }
}
  </div>
</FormattedMessageSubject>
`;

exports[`mountWithIntl using contextTypes should match the snapshot 1`] = `
<ContextTypesSubject
  foo="bar"
  intl={
    Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "formats": Object {},
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralFormat": [Function],
        "getRelativeFormat": [Function],
      },
      "locale": "en",
      "messages": null,
      "now": [Function],
      "onError": [Function],
      "textComponent": "span",
      "timeZone": null,
    }
  }
>
  <div>
    TerraEnzymeIntl.helloWorld
    <button
      type="button"
    >
      TerraEnzymeIntl.buttonText
    </button>
    {
  "foo": "bar",
  "intl": {
    "locale": "en",
    "messages": {},
    "formats": {},
    "timeZone": null,
    "textComponent": "span",
    "defaultLocale": "en",
    "defaultFormats": {},
    "formatters": {}
  }
}
  </div>
</ContextTypesSubject>
`;

exports[`mountWithIntl using injectIntl should match the snapshot 1`] = `
<InjectIntl(Component)
  foo="bar"
  intl={
    Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "formats": Object {},
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralFormat": [Function],
        "getRelativeFormat": [Function],
      },
      "locale": "en",
      "messages": null,
      "now": [Function],
      "onError": [Function],
      "textComponent": "span",
      "timeZone": null,
    }
  }
>
  <Component
    foo="bar"
    intl={
      Object {
        "defaultFormats": Object {},
        "defaultLocale": "en",
        "formatDate": [Function],
        "formatHTMLMessage": [Function],
        "formatMessage": [Function],
        "formatNumber": [Function],
        "formatPlural": [Function],
        "formatRelative": [Function],
        "formatTime": [Function],
        "formats": Object {},
        "formatters": Object {
          "getDateTimeFormat": [Function],
          "getMessageFormat": [Function],
          "getNumberFormat": [Function],
          "getPluralFormat": [Function],
          "getRelativeFormat": [Function],
        },
        "locale": "en",
        "messages": null,
        "now": [Function],
        "onError": [Function],
        "textComponent": "span",
        "timeZone": null,
      }
    }
  >
    <div>
      TerraEnzymeIntl.helloWorld
      <button
        type="button"
      >
        TerraEnzymeIntl.buttonText
      </button>
      {
  "foo": "bar"
}
    </div>
  </Component>
</InjectIntl(Component)>
`;

exports[`renderWithIntl using FormattedMessage should match the snapshot 1`] = `
<div>
  <span>
    TerraEnzymeIntl.helloWorld
  </span>
  <button
    type="button"
  >
    TerraEnzymeIntl.buttonText
  </button>
  {
  "foo": "bar",
  "intl": {
    "locale": "en",
    "messages": {},
    "formats": {},
    "timeZone": null,
    "textComponent": "span",
    "defaultLocale": "en",
    "defaultFormats": {},
    "formatters": {}
  }
}
</div>
`;

exports[`renderWithIntl using contextTypes should match the snapshot 1`] = `
<div>
  TerraEnzymeIntl.helloWorld
  <button
    type="button"
  >
    TerraEnzymeIntl.buttonText
  </button>
  {
  "foo": "bar",
  "intl": {
    "locale": "en",
    "messages": {},
    "formats": {},
    "timeZone": null,
    "textComponent": "span",
    "defaultLocale": "en",
    "defaultFormats": {},
    "formatters": {}
  }
}
</div>
`;

exports[`renderWithIntl using injectIntl should match the snapshot 1`] = `
<div>
  TerraEnzymeIntl.helloWorld
  <button
    type="button"
  >
    TerraEnzymeIntl.buttonText
  </button>
  {
  "foo": "bar"
}
</div>
`;

exports[`shallowContext with shallow using FormattedMessage should match the snapshot 1`] = `
<div>
  <FormattedMessage
    id="TerraEnzymeIntl.helloWorld"
    values={Object {}}
  />
  <FormattedMessage
    id="TerraEnzymeIntl.buttonText"
    values={Object {}}
  >
    <Component />
  </FormattedMessage>
  {
  "foo": "bar"
}
</div>
`;

exports[`shallowContext with shallow using contextTypes should match the snapshot 1`] = `
<div>
  TerraEnzymeIntl.helloWorld
  <button
    type="button"
  >
    TerraEnzymeIntl.buttonText
  </button>
  {
  "foo": "bar"
}
</div>
`;

exports[`shallowContext with shallow using injectIntl should match the snapshot 1`] = `
<Component
  foo="bar"
  intl={
    Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "formats": Object {},
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralFormat": [Function],
        "getRelativeFormat": [Function],
      },
      "locale": "en",
      "messages": null,
      "now": [Function],
      "onError": [Function],
      "textComponent": "span",
      "timeZone": null,
    }
  }
/>
`;

exports[`shallowWithIntl using FormattedMessage should match the snapshot 1`] = `
<div>
  <FormattedMessage
    id="TerraEnzymeIntl.helloWorld"
    values={Object {}}
  />
  <FormattedMessage
    id="TerraEnzymeIntl.buttonText"
    values={Object {}}
  >
    <Component />
  </FormattedMessage>
  {
  "foo": "bar",
  "intl": {
    "locale": "en",
    "messages": {},
    "formats": {},
    "timeZone": null,
    "textComponent": "span",
    "defaultLocale": "en",
    "defaultFormats": {},
    "formatters": {}
  }
}
</div>
`;

exports[`shallowWithIntl using contextTypes should match the snapshot 1`] = `
<div>
  TerraEnzymeIntl.helloWorld
  <button
    type="button"
  >
    TerraEnzymeIntl.buttonText
  </button>
  {
  "foo": "bar",
  "intl": {
    "locale": "en",
    "messages": {},
    "formats": {},
    "timeZone": null,
    "textComponent": "span",
    "defaultLocale": "en",
    "defaultFormats": {},
    "formatters": {}
  }
}
</div>
`;

exports[`shallowWithIntl using injectIntl should match the snapshot 1`] = `
<Component
  foo="bar"
  intl={
    Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "formats": Object {},
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralFormat": [Function],
        "getRelativeFormat": [Function],
      },
      "locale": "en",
      "messages": null,
      "now": [Function],
      "onError": [Function],
      "textComponent": "span",
      "timeZone": null,
    }
  }
/>
`;
